### PR TITLE
[WIP] Add initial server, routes, and authentication

### PR DESCRIPTION
**Pull request by herman-rogers**
- Add initial server and routes

- Add initial client

- Add initial server status resp

- Add authentication

### CODE REVIEW

File: api/client_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	resp := Response{}
+	if err := json.NewDecoder(w.Body).Decode(&resp); err != nil {
+		t.Fatalf(err.Error())
+	}
+	if resp.Status != "Server OK" {
+		t.Errorf("Expected %v, but got %v", "Status OK", resp.Status)
+	}
+}
+
+// func TestServerReturnsUnauthorizedAccess(t *testing.T) {
```
Comment (herman-rogers) [inline api/client_test.go]: going to add this in soon

Comment (SuperIzzo) [inline api/client_test.go]: does this need to be commented

File: api/routes.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package api
+
+// ServerStatus checks the current status of the server
+const ServerStatus = "/api/status"
+
+//const FacebookAuth = "/api/"
```
Comment (SuperIzzo) [inline api/routes.go]: comment

File: api/client_test.go (hunk 1 above)
Comment (herman-rogers) [inline api/client_test.go]: sorry kind of moving at a fast pace - next branch will be FacebookAuth which will add in a FacebookAuth route(below) and this test here becomes relevant. Wanted to have one route unauthenticated just so we can get the server status - and test to see the server is awake.

**Pull request merged**